crates/ide/src/lib.rs

```diff
@@ -99,8 +99,15 @@ pub use text_edit::{Indel, TextEdit};
 
 pub type Cancelable<T> = Result<T, Canceled>;
 
+/// Configuration parameters for the analysis run.
+#[derive(Debug, Default, Clone)]
+pub struct AnalysisConfig {
+    pub disabled_diagnostics: HashSet<String>,
+}
```

Comment:
@matklad is it okay to make this config structure part of `AnalysisHost` or should it be handled differently?

Reply:
Good question! Given that we only do surface filtering at the moment, I think it's better to just pass it in as an argument to `Analysis::diagnostics`.

If, in the future, we decide to disable diagnostics on a more fundamental level (ie, avoid computing them at all), I think we'll have to make this an input of the database. 

But for now, adding an argument seems like a better approach.

Reply:
Sure, will do that.

Reply:
Done.